lib/collection/src/collection.rs

```diff
@@ -1130,6 +1130,38 @@ impl Collection {
         Ok(())
     }
 
+    /// Updates vectors config:
+    /// Saves new params on disk
+    ///
+    /// After this, `recreate_optimizers_blocking` must be called to create new optimizers using
+    /// the updated configuration.
+    pub async fn update_vectors_from_diff(
+        &self,
+        update_vectors_diff: UpdateVectorsConfig,
+    ) -> CollectionResult<()> {
+        let mut config = self.collection_config.write().await;
+
+        // For each vector, update params
+        for (vector_name, update_params) in update_vectors_diff.params_iter() {
```

Comment:
I propose to move this `for` to a separate function like `CollectionParams::update_from_diff` and remove `CollectionParams::get_vector_params_mut` function

Reply:
Done.

We now have `CollectionParams::update_vectors_from_diff`. Updating the parameters for each vector is now done through the `ConfigDiff::update` trait.